google/cloud/internal/async_long_running_operation.h

```diff
@@ -147,6 +147,27 @@ future<StatusOr<ReturnType>> AsyncLongRunningOperation(
       });
 }
 
+/**
+ * Asynchronously polls an already started long-running operation.
+ */
+template <typename ReturnType>
+future<StatusOr<ReturnType>> AsyncAwaitLongRunningOperation(
+    google::cloud::CompletionQueue cq, ImmutableOptions options,
+    google::longrunning::Operation const& operation,
+    AsyncPollLongRunningOperation poll, AsyncCancelLongRunningOperation cancel,
+    LongRunningOperationValueExtractor<ReturnType> value_extractor,
+    std::unique_ptr<PollingPolicy> polling_policy, char const* location) {
+  using ::google::longrunning::Operation;
+  auto loc = std::string{location};
+  return AsyncPollingLoop(std::move(cq), std::move(options),
+                          make_ready_future(StatusOr<Operation>(operation)),
```

Comment:
I think you should add an overload to `AsyncPollingLoop::Start()` that consumes the operation directly, that is, take this code:

https://github.com/googleapis/google-cloud-cpp/blob/dfed337b942b940b827782464a5331319c2e519f/google/cloud/internal/async_polling_loop.cc#L49

And include something like:

```cc
  future<StatusOr<Operation>> Start(future<StatusOr<Operation>> op) {
    auto self = shared_from_this();
    return op.then([self](auto f) { return self->Start(op.get(); }
  }

  future<StatusOr<Operation>> Start(StatusOr<Operation> op) {
    auto self = shared_from_this();
    auto w = WeakFromThis();
    promise_ = promise<StatusOr<Operation>>([w, c = CallContext{}]() mutable {
      if (auto self = w.lock()) {
        ScopedCallContext scope(std::move(c));
        self->DoCancel();
      }
    });
    self->OnStart(f.get());
    return promise_.get_future();
  }
```

Eventually we should reimplement `AsyncLongRunningOperation()` using `AsyncAwaitLongRunningOperation`()` and remove the existing `AsyncPollingLoop()` overload.

Something like:

```cc
future<StatusOr<ReturnType>> AsyncLongRunningOperation(
    google::cloud::CompletionQueue cq, ImmutableOptions options,
    RequestType&& request, StartFunctor&& start,
    AsyncPollLongRunningOperation poll, AsyncCancelLongRunningOperation cancel,
    LongRunningOperationValueExtractor<ReturnType> value_extractor,
    std::unique_ptr<RetryPolicyType> retry_policy,
    std::unique_ptr<BackoffPolicy> backoff_policy, Idempotency idempotent,
    std::unique_ptr<PollingPolicy> polling_policy, char const* location) {
  using ::google::longrunning::Operation;
  auto operation =
      AsyncRetryLoop(std::move(retry_policy), std::move(backoff_policy),
                     idempotent, cq, std::forward<StartFunctor>(start), options,
                     std::forward<RequestType>(request), location);
  auto loc = std::string{location};
  return operation.then([=](auto f) mutable {
    return AsyncAwaitLongRunningOperation(
       std::move(cq), std::move(options),
       f.get(), // note we pass the `StatusOr<T>` and not a future
       std::move(poll),
       std::move(cancel), std::move(polling_policy),
       std::move(location))
  });
}
```